src/DateRangePicker.ts

```diff
@@ -1,26 +1,452 @@
+  @property({ type: Number }) _leftSliderX: number = 0;
+  @property({ type: Number }) _rightSliderX: number = 0;
+
+  @query('#tooltip') tooltip!: HTMLDivElement;
+  @query('#container') container!: HTMLDivElement;
+
+  // these properties don't need to be tracked for changes
+  _minDate: number = 0;
+  _maxDate: number = 0;
+  _dragOffset: number = 0;
+  _histWidth: number = 0;
+  _numBins: number = 0;
+  _binWidth: number = 0;
+  _currentSlider?: SVGRectElement;
+  _histData: HistogramItem[] = [];
+
+  /* eslint-enable lines-between-class-members */
+
   static styles = css`
-    :host {
-      display: block;
-      padding: 25px;
-      color: var(--date-range-picker-text-color, #000);
+    #container {
+      margin: 0;
+      touch-action: none;
+      position: relative;
+    }
+    /* prevent selection from interfering with tooltip, especially on mobile */
+    /* https://stackoverflow.com/a/4407335/1163042 */
+    .noselect {
+      -webkit-touch-callout: none; /* iOS Safari */
+      -webkit-user-select: none; /* Safari */
+      -moz-user-select: none; /* Old versions of Firefox */
+      -ms-user-select: none; /* Internet Explorer/Edge */
+      user-select: none; /* current Chrome, Edge, Opera and Firefox */
+    }
+    .bar:hover {
+      fill-opacity: 0.7;
+    }
+    /****** histogram ********/
+    #tooltip {
+      position: absolute;
+      background: ${tooltipBackgroundColor};
+      color: ${tooltipTextColor};
+      text-align: center;
+      border-radius: 3px;
+      padding: 2px;
+      font-size: ${tooltipFontSize};
+      font-family: sans-serif;
+      touch-action: none;
+      pointer-events: none;
+    }
+    #tooltip:after {
+      content: '';
+      position: absolute;
+      margin-left: -5px;
+      top: 100%;
+      /* arrow */
+      border: 5px solid ${tooltipTextColor};
+      border-color: ${tooltipBackgroundColor} transparent transparent
+        transparent;
+    }
+    /****** slider ********/
+    .draggable:hover {
+      cursor: grab;
+    }
+    .dragging {
+      cursor: grabbing !important;
+    }
+    /****** inputs ********/
+    #inputs {
+      display: flex;
+      justify-content: center;
+    }
+    #inputs .dash {
+      position: relative;
+      bottom: 1px;
+    }
+    input {
+      width: ${inputWidth};
+      margin: 0 3px;
+      border: ${inputBorder};
+      border-radius: 2px !important;
+      text-align: center;
+      font-size: ${inputFontSize};
     }
   `;
 
-  @property({ type: String }) title = 'Hey there';
+  firstUpdated(): void {
+    this._leftSliderX = this.sliderWidth;
+    this._rightSliderX = this.width - this.sliderWidth;
+    this._minDate = dayjs(this.data?.minDate).valueOf();
+    this._maxDate = dayjs(this.data?.maxDate).valueOf();
+    this._histWidth = this.width - this.sliderWidth * 2;
+    this._numBins = this.data?.bins?.length ?? 1;
+    this._binWidth = this._histWidth / this._numBins;
+    this._histData = this.generateHistData();
```

Comment:
If the reactive approach below doesn't pan out, you can make `_histData` automatically update whenever `data` gets updated by doing this inside the `updated(props: ProperyValues)` lifecycle instead of `firstUpdated`:
```
updated(changedProps: PropertyValues): void {
  if (changedProps.has('data')) {
    this._histData = this.generateHistData();
  }
}
```
Ideally, the properties dependent on `data` should trigger updates downstream so if you change it, the widget updates itself.